Update ratos_hybrid_corexy.py

dbstf opened the pull request:
Update to make compatible with Klipper refactoring (https://github.com/Klipper3d/klipper/pull/6815), in particular
* idex_modes: Internal refactoring of handling dual carriages

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Added the ability to clear homing state for specific axes, enabling targeted re-homing without affecting others.

- **Improvements**
  - Improved dual-carriage handling and homing coordination for more consistent multi-carriage movement.
  - Refined homing logic and parameter handling to reduce unexpected behavior during homing and position setting.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

dbstf committed:
Update ratos_hybrid_corexy.py

Update to make compatible with Klipper refactoring (https://github.com/Klipper3d/klipper/pull/6815), in particular
* idex_modes: Internal refactoring of handling dual carriages

tg73 commented:
Hi Stefan @dbstf - can I ask how much practical testing this change has received and on what machines?

dbstf commented:
Hi Tom, unfortunately, the code is not tested in production at all, yet. I should have mentioned this in the original commit. I'm currently in the process of piecing all parts together for a V-Core 4. For testing, I flashed the latest release of RatOS and updated Klipper to the recent version on Github. The most recent updates broke the ratos_kinematic (lots of warning on the Mainsail status screen), hence I updated the script to the best of my knowledge till all errors in Mainsail were gone. I committed my modified script to provide a starting point for all the other people developing, updating and testing RatOS.

tg73 commented:
If you're going to be commissioning a new machine, I'd recommend that you stick with the pinned commit of klipper that's defined in the `RatOS/moonraker.conf` file rather than manually pulling the latest version via git (which I assume is what you did). We have the pinned commit because we have to be rather careful when moving to a new klipper version as things often break. Moving beyond the pinned commit is only for the brave (and ideally experienced), and generally only worth considering if there's some specific klipper fix or feature that you need that comes after the pinned commit. If you do decide to move beyond the pinned commit, be sure to mention that if you run into any kind of problem and need to ask for help on discord or facebook etc.

dbstf committed:
Update ratos_hybrid_corexy.py

Update to make compatible with https://github.com/Klipper3d/klipper/commit/9399e738bc21e70cdd5a60bc9f8e1d51fb20cd75 ,i.e., https://github.com/Klipper3d/klipper/pull/7001
